iaso/api/org_units.py

```diff
@@ -763,9 +763,14 @@ def create(self, _, request):
 
     def retrieve(self, request, pk=None):
         org_unit: OrgUnit = get_object_or_404(
-            self.get_queryset().prefetch_related("reference_instances").annotate(instances_count=Count("instance")),
+            self.get_queryset().prefetch_related("reference_instances"),
             pk=pk,
         )
+
+        if request.query_params.get("instances_count"):
```

Comment:
@kemar  I used that param because the retrieve is used somewhere else(to avoid the performance issue) and as you can see the as_dict_with_parents the instances_count is optianal.
![Screenshot from 2024-11-08 12-44-17](https://github.com/user-attachments/assets/564ba030-c201-4735-9ab2-9eb513dd28ae)

Reply:
When it has to count many instances for a parent orgUnit which has many descendants

Reply:
I think the additional aggregate query is quite performant.

The point is that is API is already really ugly and hard to understand.

Here, you're adding a new parameter and it's "hidden" somewhere in the `retrieve()` method which adds to the global complexity.

I'd rather remove it until we're certain we have a performance issue.